app/controllers/lists_controller.rb

```diff
@@ -11,6 +11,9 @@ def index
   # GET /lists/1
   # GET /lists/1.json
   def show
+      @references = @list.references
+      .joins(:paper)
+      .order(('cached_votes_up DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
```

Comment:
Please never do this `('cached_votes_up DESC, ' if params[:sort] != 'pub_date').to_s` in the middle of a function call if you can avoid it.